src/module/engine/chat/chatbox.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Helper class to handle the display of chatBox
+ */
+export class SFRPGCustomChatMessage {
+
+    static getAmmoLeft(itemData){
+        if(itemData.capacity.value > 0) {
+            const finalAmmo = itemData.capacity.value - itemData.usage.value;
+            return ((finalAmmo) >= 0) ? finalAmmo : 0;
+        }
+        return null;
+    }
+
+    static hasCapacity(itemData){
+        return itemData.capacity > 0 ? itemData.capacity : 0;
+    }
+
+    static createToken(actor){
+        if(actor.token){
+            return actor.token.scene.data._id + '.' +actor.token.data._id;
+        }else if(canvas.tokens.controlled[0].id){
+            return game.scenes.active._id + '.' + canvas.tokens.controlled[0].id;
+        }else {
+            return {}
+        }
+    }
+
+    static rollToRender(roll, data, action){
+        //Get the template
+        const temmplateName = "systems/sfrpg/templates/chat/chat-message-attack-roll.html";
+        //get Actor
+        const actor = (data.actor) ? data.actor : {};
+        const item = (data.data.item) ? data.data.item : {};
+        //Render the roll
+        const customRoll = roll.render();
+        customRoll.then((diceRoll) => {
+
+            const targetTemplate = renderTemplate(temmplateName, {
+                    hasAttack: (item.hasAttack) ? item.hasAttack : false,
+                    hasDamage: (item.hasDamage) ? item.hasDamage : false,
+                    isVersatile: (item.isVersatile) ? item.isVersatile : false,
+                    hasSave: (item.hasSave) ? item.hasSave : false,
+                    title: (data.title) ? data.title : 'Roll',
+                    rawTitle: (data.speaker.alias),
+                    item: (item) ? item : {},
+                    ammoLeft: item.hasCapacity ? this.getAmmoLeft(item.data) : null,
+                    flavor: (data.flavor) ? data.flavor : '',
+                    dataRoll: roll,
+                    type: CHAT_MESSAGE_TYPES.ROLL,
+                    config: CONFIG.STARFINDER,
+                    tokenImg: actor.data.token.img,
+                    actorId: actor._id,
+                    tokenId: this.createToken(actor),
+                });
+
+                if(data.speaker.alias){
```

Comment:
`if` statement should have a space between the term `if` and the opening parenthesis, and then a space between the closing parenthesis and the opening curly bracket; like `if (condition) {`.

Reply:
done